engine/src/main/java/io/zeebe/engine/state/EventAppliers.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.state;
+
+import io.zeebe.engine.state.appliers.WorkflowInstanceElementActivatedApplier;
+import io.zeebe.msgpack.UnpackedObject;
+import io.zeebe.protocol.impl.record.RecordMetadata;
+import io.zeebe.protocol.record.ValueType;
+import io.zeebe.protocol.record.intent.Intent;
+import io.zeebe.protocol.record.intent.WorkflowInstanceIntent;
+import java.util.Map;
+
+/**
+ * Applies state changes from events to the {@link io.zeebe.engine.state.ZeebeState}.
+ *
+ * <p>Finds the correct {@link EventApplier} and delegates.
+ */
+public final class EventAppliers {
+
+  private static final EventApplier UNIMPLEMENTED_EVENT_APPLIER =
+      (key, intent, value, metadata) -> {};
+
+  private final Map<ValueType, Map<Intent, EventApplier>> mapping;
+
+  public EventAppliers(final ZeebeState state) {
```

Comment:
Maybe move the class into the appliers package. This could also give you new options with package visibility to hide the individual `EventApplier` implementations (e.g. package constructor for the implementations of `EventApplier`)

Reply:
Agreed 👍